src/utils/url.utils.ts

```diff
@@ -104,3 +111,57 @@ export const getWebsocketURL = () => {
 
     return `wss://${serverURL}/websockets/v3?app_id=${getAppId()}&l=${language}&brand=${AppIDConstants.appBrand}`;
 };
+
+/**
+ * Extracts query parameters from the URL by parsing the current window's URL search parameters for the specified key.
+ * It returns the query parameters associated with the given key.
+ *
+ * @returns {string | null} A string containing query parameter associated with the given key.
+ */
+export const getURLParameters = (key: queryParameters) => {
+    const searchParams = new URLSearchParams(window.location.search);
+    return searchParams.get(key);
+};
+
+/**
+ * Takes a 'URL path' as input and removes certain characters from the beginning and end of the path.
+ *
+ * Removes the following:
+ * - Any leading forward slash (/) at the beginning of the path.
+ * - Any trailing forward slash (/) at the end of the path.
+ * - Any characters that are not alphanumeric, hyphen, underscore, dot, forward slash, parentheses, or hash symbol.
+ *
+ * @param {string} path - The URL path that needs to be normalized.
+ * @returns {string} Returns the formatted path without the specified characters.
+ */
+export const normalizePath = (path: string) => path.replace(/(^\/|\/$|[^a-zA-Z0-9-_./()#])/g, "");
+
+/**
+ * Generates a static URL for the deriv.com project based on the provided parameters.
+ * This function is necessary because deriv.com URLs differ from those used in app.deriv.com
+ *
+ * @param {string} path - The path to be appended to the base URL.
+ * @param {boolean} [isDocument=false] - Specifies whether the path represents a document.
+ * @param {boolean} [isEU=false] - Specifies whether the URL should be generated for the EU production environment.
+ * @returns {string} Returns the formatted static URL.
+ */
+export const generateDerivStaticURL = (path: string, isDocument?: boolean, isEU?: boolean) => {
+    const { derivComProductionEU, derivComProduction } = URLConstants;
+    const { i18nLanguage } = LocalStorageConstants;
+
+    const host = isEU ? derivComProductionEU : derivComProduction;
+    let lang = localStorage.getItem(i18nLanguage)?.toLowerCase();
```

Comment:
```suggestion
    let lang = localStorage.getItem(LocalStorageConstants.i18nLanguage)?.toLowerCase() ?? "en";
```
it is better to do this since now it is clear the `i18nLanguage`constant variable belongs in the `LocalStorageConstants` namespace. Defaults to "en" this way we can skip falsy check in the next line